### PR TITLE
Fix test-erasure-eio and osd-scrub-repair races (17830)

### PR DESCRIPTION
**Pull request by dzafman**
I want Jenkins to run this.  And probably have @tchaikov cherry-pick the test commit to ceph/ceph#11914 and just close this pull request.

### CODE REVIEW

**Comment by tchaikov** [comment]
@dzafman, i pull your PR in once i figure out why the jenkins "make check" still fails and fix it.

**Comment by dzafman** [comment]
@tchaikov My changes in osd-scrub-repair.sh might fix the races.

**Comment by dzafman** [comment]
Jenkins passed, but let's try another attempt.  I have a commit that fixes compiler warnings.

**Comment by dzafman** [comment]
@athanatos If you want to move these tests to teuthology we don't have to put this pull request in.  It seems to fix the races I know about, but can't be sure there aren't others.

**Comment by athanatos** [comment]
@dzafman Meh, if it's reliable, it's not a particularly high priority to move it to teuthology.

**Comment by athanatos** [comment]
Looks like jenkins failed though.

**Comment by dzafman** [comment]
I fixed something else in osd-scrub-repair.  Let's try it again.

retest this please

**Comment by dzafman** [comment]
@athanatos @tchaikov Seemingly unrelated unittest_journal failed.

File: src/test/erasure-code/test-erasure-code-plugins.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ function run() {
     local dir=$1
     shift
 
-    export CEPH_MON="127.0.0.1:17109"
+    export CEPH_MON="127.0.0.1:17110" # git grep '\<17110\>' : there must be only one
```
Comment (tchaikov) [inline src/test/erasure-code/test-erasure-code-plugins.sh]: oh my! good catch!

**Pull request merged**